pkg/sql/backfill/backfill.go

```diff
@@ -366,19 +379,65 @@ func (ib *IndexBackfiller) Init(
 		if IndexMutationFilter(m) {
 			idx := m.GetIndex()
 			ib.added = append(ib.added, *idx)
-			for i := range cols {
-				id := cols[i].ID
+			for i := range ib.cols {
+				id := ib.cols[i].ID
 				if idx.ContainsColumnID(id) ||
 					idx.GetEncodingType(desc.PrimaryIndex.ID) == sqlbase.PrimaryIndexEncoding {
 					valNeededForCol.Add(i)
 				}
 			}
+
+			// If the index is a partial index, convert and store the predicate
+			// as a TypedExpr.
+			if idx.IsPartial() {
```

Comment:
Can you take a look at that changes made to the ColumnBackfiller in https://github.com/cockroachdb/cockroach/pull/51865 -- that is the end state for what i imagine these backfillers will look like when there are user defined types in these expressions. Right now, this wouldn't work with user defined types. I'm also OK with it not handling it right now and then opening an issue for it to handle later

Reply:
Nice, looks like that should be pretty straightforward to add. I've added a note in [the issue I'm using to track partial indexes + UDTs](https://github.com/cockroachdb/cockroach/issues/51804). I'll probably go through in one swoop and add a bunch of UDT+partial index tests and fix whatever is broken, if that's ok with you. There's a number of existing bugs at the moment.

Reply:
Yup, seems reasonable to me.